src/OpenTelemetry.OpAMPClient/Listeners/Messages/AgentIdentificationMessage.cs

```diff
@@ -0,0 +1,11 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using Opamp.Protocol;
+
+namespace OpenTelemetry.OpAMPClient.Listeners.Messages;
+
+internal record AgentIdentificationMessage : IOpAMPMessage
+{
+    public required AgentIdentification AgentIdentification { get; set; }
```

Comment:
I'm thinking in a second iteration those should be refactored to public dtos, so any plugins can handle / listen for any message. CustomMessages need anyway extendability.